### ISSUE TITLE
Segfault with WebSocket

### ISSUE THREAD

**Issue by o3o** [rejectedsoftware/vibe.d#1384]
After update vibe.d to v0.7.27-beta.1-7-g5dff616, suddenly my app crash.

I use WebSocket, and the problem is on 
```
void getWS(int ct, string action, scope WebSocket socket) {
  auto t = runTask({
       logInfo("CONN %s", trayNum);
       while (socket.connected) {
          ....
       }
  });
  
  while (socket.waitForData) {   // <- issue is here
       ....
  }
  t.join();
}
```
The error was generated in `source/vibe/core/drivers/libevent2_tcp.d` line 297
```
  if (dataAvailableForRead) return true;
  if (m_ctx.state != ConnectionState.open) return false;
```
If I add an `assert`:
```
  if (dataAvailableForRead) return true;
  assert(m_ctx !is null, "mctx is NULL!!!");
  if (m_ctx.state != ConnectionState.open) return false;
```

 I get this message:
```
Task terminated with unhandled exception: mctx is NULL!!!
core.exception.AssertError@source/vibe/core/drivers/libevent2_tcp.d(302): mctx is NULL!!!
```

I solved my problem with this patch:
```
                if (dataAvailableForRead) return true;
-               if (m_ctx.state != ConnectionState.open) return false;
+               if ((m_ctx is null) || m_ctx.state != ConnectionState.open) return false;
 
                acquireReader();
```

Thank you

**Comment by s-ludwig** [comment]
Fixed by 4b192bde (I hit the same issue yesterday). Thanks for investigating!

**Comment by o3o** [comment]
Thank to you for your great job.
